src/program.js

```diff
@@ -450,6 +459,13 @@ Example: $0 --help run.
       type: 'boolean',
       demandOption: false,
     },
+    'input': {
+      // This option is defined to make yargs to accept the --no-input
+      // defined above, but we hide it from the yargs help output.
```

Comment:
Could you add a unit test to verify that it doesn't appear in the help?

Reply:
with an integration test for sure, but I'll take a look if we can cover it with a unit test first.

Reply:
added as a new functional test case (it would be possible to unit test if we would have access to the `Program` instance created internally by the `main` function, which would require a bit of refactoring, I opted for the function test as it was straightforward and our small functional tests are still pretty fast to run).